core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/kubernetesModels.scala

```diff
@@ -158,13 +162,50 @@ final case class Nodepool(id: NodepoolLeoId,
                           autoscalingEnabled: Boolean,
                           autoscalingConfig: Option[AutoscalingConfig],
                           errors: List[KubernetesError],
-                          apps: List[App])
+                          apps: List[App],
+                          isDefault: Boolean)
 object KubernetesNameUtils {
   //UUID almost works for this use case, but kubernetes names must start with a-z
   def getUniqueName[A](apply: String => A): Either[Throwable, A] =
     KubernetesName.withValidation('k' + UUID.randomUUID().toString.toLowerCase.substring(1), apply)
 }
 
+case class DefaultNodepool(id: NodepoolLeoId,
```

Comment:
This class seems a bit duplicative of `Nodepool`. Is there a strong need for a separate type given we also have `Nodepool.isDefault`?

Reply:
I think it is helpful to have to be able to enforce the type of nodepool being passed around with type safety, as opposed to validating a boolean. Without it, there are some scenarios where you need to specify Apps/Errors and it doesn't make sense.

The use case for the boolean is to filter db results, which have no notion of the ADT differentiation